src/decoder/lattice-faster-decoder.cc

```diff
@@ -78,7 +78,7 @@ void LatticeFasterDecoderTpl<FST, Token>::InitDecoding() {
 template <typename FST, typename Token>
 bool LatticeFasterDecoderTpl<FST, Token>::Decode(DecodableInterface *decodable) {
   InitDecoding();
-
+  AdvanceDecoding(decodable);
```

Comment:
Please remove the while loop below this; it's now redundant because it's done in AdvanceDecoding().
And see if there are any other decoders templated in the same way that have the same issue (grep for Decode).